### DISCUSSION TITLE
Send Emails Using the Lib

### DISCUSSION THREAD

**Discussion by GabrielFMPinheiro** [AleksandrRogov/DynamicsWebApi#173] [Q&A]
Does anyone know how to send emails with parameters and attachments using a specific template with this library? The documentation is not very clear on how to perform this operation through the backend.

**Comment by AleksandrRogov** [comment]
Hi @GabrielFMPinheiro , you just need to know what you need to do in Dynamics itself and just re-create the steps using the web api (with help of the library):

1. Create an email activity (`emails`)
2. If your email has a body (you have template, so won't need this step, I think) - create a "Note" record (`annotations`) and assign it to a created email in step 1 using the "Regarding" navigation property.
3. Create an email attachment. Create an "ActivityMimeAttachment" record `activitymimeattachments` and assign it to an email created in step 1 using the "Regarding" navigation property.
4. You must already have a template, in this case you must already have an ID of that template stored somewhere in environment variables of your application, or dynamically fetched with your current UI.
5. You must already have a recipient of the email somewhere in your app. Again, either stored somewhere or fetch with your UI.
6. Execute `SendEmailFromTemplate` [action](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/sendemailfromtemplate?view=dataverse-latest). With parameters: 
`Target` - email created in step 1
`Regarding` - recipient of the email in step 5.
`TemplateId` - the id of the template in step 4.

Hope this helps!

**Reply by GabrielFMPinheiro** [reply]
Hey @AleksandrRogov ! Thank you! I will try to follow these steps.
